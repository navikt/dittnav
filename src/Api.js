import { Dittnav, Innlogging, TestProducer } from './constants';

export const redirectToLogin = () => {
  window.location.assign(`${Innlogging.LOGINSERVICE_URL}`);
};

export const tokenExpiresSoon = (headers) => (
  headers.get('x-token-expires-soon')
);

const fetchJSON = (url) => new Promise((resolve, reject) => {
  fetch(url, { method: 'GET', credentials: 'include' })
    .then(response => {
      if (response.ok) {
        return response.json()
          .then(json => [json, response.headers]);
      }
      reject(response);
      return null;
    })
    .then(([content, headers]) => {
      resolve({ content, headers });
    })
    .catch(e => reject(e));
});

export const checkAuth = () => new Promise((resolve, reject) => {
  fetchJSON(`${Innlogging.INNLOGGINGSSTATUS_URL}?ts=${Date.now()}`)
    .then(({ content }) => {
      if (content.authenticated) {
        resolve(content);
      } else {
        reject(new Error('not authenticated'));
      }
    })
    .catch(() => reject(new Error('not authenticated')));
});

export const checkApiStatus = () => new Promise((resolve, reject) => {
  fetchJSON(`${Dittnav.API_AUTH_URL}`)
    .then(([response]) => {
      resolve(response);
    })
    .catch(e => {
      reject(e);
    });
});

const postJSON = (url, content) => new Promise((resolve, reject) => {
  fetch(url, {
    method: 'POST',
    credentials: 'include',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(content),
  })
    .then(response => response.headers)
    .then((headers) => resolve(headers))
    .catch((e) => reject(e));
});

export const fetchOppfolging = () => (
  fetchJSON(Dittnav.OPPFOLGING_URL)
);

export const fetchMeldekort = () => (
  fetchJSON(Dittnav.MELDEKORT_URL)
);

export const fetchPersonNavn = () => (
  fetchJSON(Dittnav.PERSON_NAVN_URL)
);

export const fetchPersonIdent = () => (
  fetchJSON(Dittnav.PERSON_IDENT_URL)
);

export const fetchSaker = () => (
  fetchJSON(Dittnav.SAKER_URL)
);

export const fetchMeldinger = () => (
  fetchJSON(Dittnav.MELDINGER_URL)
);

export const fetchSakstema = () => (
  fetchJSON(Dittnav.SAKSTEMA_URL)
);

export const fetchInnloggingsstatus = () => (
  fetchJSON(Innlogging.INNLOGGINGSSTATUS_URL)
);

export const fetchBeskjeder = () => (
  fetchJSON(Dittnav.BESKJED_URL)
);

export const fetchOppgaver = () => (
  fetchJSON(Dittnav.OPPGAVE_URL)
);

export const fetchInnbokser = () => (
  fetchJSON(Dittnav.INNBOKS_URL)
);

export const fetchInaktiveBeskjeder = () => (
  fetchJSON(Dittnav.BESKJED_INAKTIV_URL)
);

export const fetchInaktiveOppgaver = () => (
  fetchJSON(Dittnav.OPPGAVE_INAKTIV_URL)
);

export const fetchInaktiveInnbokser = () => (
  fetchJSON(Dittnav.INNBOKS_INAKTIV_URL)
);

export const postHendelse = (path, content) => (
  postJSON(`${TestProducer.URL}/${path}`, content)
);

export const postDoneAll = () => (
  postJSON(`${TestProducer.DONE_ALL_URL}`, null)
);

export const postDone = (content) => (
  postJSON(`${Dittnav.DONE_URL}`, content)
);

export const postStatusoppdatering = (content) => (
  postJSON(`${TestProducer.STATUSOPPDATERING_URL}`, content)
);
