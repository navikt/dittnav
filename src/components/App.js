import React, { Component } from 'react';
import PersonInfo from './PersonInfo';
import InfoMeldinger from './InfoMeldinger';
import Tjenester from './Tjenester';
import Lenkelister from './Lenkelister';
import Artikler from './Artikler';

// // dittnav\web\src\main\java\no\nav\sbl\dialogarena\dittnav\web\pages\ForsideInnlogget.html
class App extends Component {
  render() {
    return (
      <main role="main">
        <section className="service-notification" style={{ display: 'block' }}>
          <h1 className="typo-element">
            <span style={{ display: 'inline' }}>Det skjedde en feil under henting av saker</span>
            <span style={{ display: 'inline' }}>Det skjedde en feil under henting av meldinger fra din innboks</span>
          </h1>
        </section>
        <div className="container">
          <div className="row">
            <div className="maincontent side-innhold">
              <div className="col-md-12">
                <PersonInfo />
                <InfoMeldinger />
                <Tjenester />
                <Lenkelister />
              </div>
            </div>
          </div>

          <Artikler />

        </div>
      </main>
    );
  }
}

export default App;
