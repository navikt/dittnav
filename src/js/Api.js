import conf from './Config';

const redirectToLogin = () => {
  window.location.assign(`${conf.dittNav.LOGINSERVICE}&redirect=${window.location.href}`); // eslint-disable-line no-undef
};

const fetchUnleashFeatures = (features) => {
  const fString = features.map(f => `feature=${f}`);
  const URL = `${conf.dittNav.CONTEXT_PATH}/api/feature`;
  return Promise.race([
    fetch(`${URL}?${fString.join('&')}`, { method: 'GET' }) // eslint-disable-line no-undef
      .then(r => r.json()),
    new Promise((_, reject) => setTimeout(() => {
      const message = `Couldnt wait for unleash longer than ${conf.UNLEASH_TIMEOUT} msec`;
      return reject(new Error(message));
    }, conf.UNLEASH_TIMEOUT))]);
};

const fetchJSONAndCheckForErrors = (url) => {
  return new Promise((res, rej) => {
    fetch(url, { method: 'GET', credentials: 'include' }) // eslint-disable-line no-undef
      .then((r) => {
        if (r.status === 401 || (r.status === 0 && !r.ok)) {
          redirectToLogin();
          return;
        }
        if (!r.ok) {
          rej(new Error('Error happened on requesting a resource'));
          return;
        }
        res(r.json());
      })
      .catch((e) => {
        rej(e);
      });
  });
};

const checkAuth = () => {
  return new Promise((res, rej) => {
    fetchJSONAndCheckForErrors(`${conf.INNLOGGINGSLINJE_AUTH}`)
      .then((r) => {
        if (!r.authenticated) {
          redirectToLogin();
          return;
        }
        res(r);
      })
      .catch((e) => {
        rej(e);
      });
  });
};

const sendJSONAndCheckForErrors = (tekst, url = `${conf.dittNav.DITTNAV_HENDELSER_URL}`) => {
  fetch(url, {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      tekst,
      link: 'https://localhost/100',
    }),
  })
    .then((r) => r.status)
    .catch((e) => console.log(`ERROR: ${e}`));
};

const fetchPersonInfoAndServices = () => fetchJSONAndCheckForErrors(`${conf.dittNav.DITTNAV_API_URL}`);
const fetchSaker = () => fetchJSONAndCheckForErrors(`${conf.dittNav.DITTNAV_SAKER_URL}`);
const fetchMeldinger = () => fetchJSONAndCheckForErrors(`${conf.dittNav.DITNTAV_MELDINGER_URL}`);
const fetchSakstema = () => fetchJSONAndCheckForErrors(`${conf.dittNav.DITTNAV_SAKSTEMA_URL}`);
const fetchHendelser = () => fetchJSONAndCheckForErrors(`${conf.dittNav.DITTNAV_HENDELSER_URL}`);

export default {
  fetchUnleashFeatures,
  checkAuth,
  fetchPersonInfoAndServices,
  fetchSaker,
  fetchMeldinger,
  fetchHendelser,
  fetchSakstema,
  sendHendelser: sendJSONAndCheckForErrors,
};
