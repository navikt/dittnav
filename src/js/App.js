import React, { Component } from 'react';
import PropTypes from 'prop-types';

import FeilMeldinger from './components/FeilMeldinger';
import Home from './pages/Home';
import Hendelser from './components/Hendelser';

import '../less/index.less';
import './polyfill';

import Environments from './Environment';

const App = ({ api }) => {

  const [info, setInfo] = React.useState({});
  const [paabegynteSoknader, setPaabegynteSoknader] = React.useState(null);
  const [mininnboks, setMininnboks] = React.useState([]);
  const [errors, setErrors] = React.useState([]);
  const [fetching, setFetching] = React.useState(0);

  React.useEffect(() => {
    const catchError = msg => () => {
      errors.push(msg);
      ///this.setState(() => ({ errors, fetching: this.state.fetching + 1 }));
      setErrors(errors);
      setFetching(fetching + 1);
    };

    api.fetchPersonInfoAndServices()
      .then((r) => {
        const { feilendeTjenester } = r;
        if (feilendeTjenester.length > 0) {
          errors.push('error.baksystemer');
        }
        // this.setState(() => ({ info: r, errors, fetching: this.state.fetching + 1 }));
        setInfo(r);
        setErrors(errors);
        setFetching(fetching + 1);
      })
      .catch(catchError('error.baksystemer'));

    api.fetchSaker()
      .then((r) => {
        const { feilendeBaksystem }Â = r;
        if (feilendeBaksystem.length > 0) {
          errors.push('error.baksystemer');
        }
        // this.setState(() => ({ paabegynteSoknader: r, fetching: this.state.fetching + 1 }));
        setPaabegynteSoknader(r);
        setFetching(fetching + 1);
      }).catch(catchError('error.baksystemer'));

    api.fetchMeldinger()
      .then((r) => {
        //this.setState(() => ({ mininnboks: r, fetching: this.state.fetching + 1 }));
        setMininnboks(r)
        setFetching(fetching + 1)
      }).catch(catchError('error.baksystemer'));
  }, []);

    const uniqueErrors = errors.filter((item, i, ar) => ar.indexOf(item) === i);
    const erIDev = Environments() === 'DEV';

    return (
      <main role="main">
        <FeilMeldinger errors={uniqueErrors} />
        <div className="container">
          { erIDev ? <Hendelser /> : null }
          <Home info={info} paabegynteSoknader={paabegynteSoknader} mininnboks={mininnboks} fetching={fetching} />
        </div>
      </main>
    );
}

App.propTypes = {
  api: PropTypes.shape({
    fetchPersonInfoAndServices: PropTypes.func.isRequired,
    fetchSaker: PropTypes.func.isRequired,
    fetchMeldinger: PropTypes.func.isRequired,
  }).isRequired,
};

export default App;
