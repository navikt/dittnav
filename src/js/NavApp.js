import React, { Component } from 'react';

import { IntlProvider, addLocaleData } from 'react-intl';

import nb from 'react-intl/locale-data/nb';
import en from 'react-intl/locale-data/en';
import nn from 'react-intl/locale-data/nn';

addLocaleData([...nb, ...en, ...nn]);

class NavApp extends Component {
  render() {
    return (
      <IntlProvider locale={this.props.defaultSprak} messages={this.props.messages[this.props.defaultSprak]}>
        {this.props.children}
      </IntlProvider>
    );
  }
}

export default NavApp;
