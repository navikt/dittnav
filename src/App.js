import React, { useEffect } from 'react';
import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import { useQuery } from 'react-query';
import { Toggle } from './constants';
import useApiQueries from './hooks/useApiQueries';
import ScrollToTop from './components/scroll/ScrollToTop';
import Home from './pages/Home/Home';
import Varslinger from './pages/Varslinger/Varslinger';
import HendelserTestSide from './pages/Hendelser/HendelserTestSide';
import Statusoppdatering from './pages/Statusoppdatering/Statusoppdatering';
import Tidslinje from './pages/Tidslinje/Tidslinje';
import { fetchUnleashToggle } from './Api';

const App = () => {
  const { data: minSideToggle } = useQuery('minside', fetchUnleashToggle);
  useApiQueries();

  useEffect(() => {
    if (minSideToggle?.content === true) {
      window.location.assign(`${window.env.NAVNO_URL}/minside`);
    }
  }, [minSideToggle]);

  return (
    <Router basename="/person">
      <ScrollToTop />
      <Switch>
        <Route
          path="/dittnav"
          exact
          component={Home}
        />
        <Route
          path="/dittnav/varslinger"
          exact
          component={Varslinger}
        />
        {Toggle.TEST_SIDE && (
          <Route
            path="/dittnav/hendelser"
            exact
            component={HendelserTestSide}
          />
        )}
        {Toggle.TEST_SIDE && (
          <Route
            path="/dittnav/statusoppdatering"
            exact
            component={Statusoppdatering}
          />
        )}
        {Toggle.TEST_SIDE && (
          <Route
            path="/dittnav/tidslinje"
            exact
            component={Tidslinje}
          />
        )}
        <Route
          path="/dittnav*"
          exact
          component={Home}
        />
      </Switch>
    </Router>
  );
};

export default App;
