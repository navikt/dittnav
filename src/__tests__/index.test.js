import React from 'react';
const ReactTestRenderer = require('react-test-renderer');
import App from '../js/App';
import Api from '../js/Api';
import NavApp from '../js/NavApp';

import nbMessages from 'translations/nb.json';
import enMessages from 'translations/en.json';

const loadMessages = () => {
  return {
    nb: nbMessages,
    en: enMessages,
  };
};

const wrapNavApp = (children, props = { defaultSprak: 'nb', messages: loadMessages() }) => (
  <NavApp {...props}>
    {children}
  </NavApp>
);

it('index renders without crashing', () => {
  const component = ReactTestRenderer.create(wrapNavApp(<App api={Api} />));
  expect(component.toJSON()).toMatchSnapshot();
});
