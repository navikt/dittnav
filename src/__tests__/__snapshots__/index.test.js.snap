// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders without crashing 1`] = `
<main
  role="main"
>
  <div
    className="container"
  >
    <div
      className="Hendelser"
    >
      <h1
        className="typo-innholdstittel"
      >
        <span>
          Opprettelse og visning av hendelser
        </span>
      </h1>
      <form
        onSubmit={[Function]}
      >
        <div
          className="skjemaelement"
        >
          <label
            className="skjemaelement__label"
            htmlFor="655514698-2985-3988-10168-068790813904"
          >
            Skriv inn ny tekst:
          </label>
          <input
            className="skjemaelement__input input--fullbredde"
            id="655514698-2985-3988-10168-068790813904"
            onChange={[Function]}
            type="text"
            value=""
          />
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
        <div
          className="Knapper"
        >
          <button
            className="knapp knapp--standard"
            disabled={false}
            type="submit"
          >
            <span>
              Send hendelse
            </span>
          </button>
          <button
            className="knapp knapp--standard"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span>
              Hent hendelser
            </span>
          </button>
        </div>
      </form>
      <div
        className="InformasjonHendelser"
      />
    </div>
    <div
      className="row"
    >
      <div
        className="maincontent side-innhold"
      >
        <div
          className="col-md-12"
          id="dittnav-main-container"
        >
          <section
            className="infomeldinger-list"
          >
            <h1
              className="vekk"
            >
              <span>
                Varsler
              </span>
            </h1>
          </section>
          <div
            className="dittnav-lenke-container blokk-xl"
          >
            <a
              className="lenkepanel first lenkepanel--border"
              href="http://localhost:9222/utbetalingsoversikt/"
            >
              <span
                className="typo-undertittel lenkepanel__heading"
              >
                <span>
                  Dine utbetalinger
                </span>
              </span>
              <span
                className="lenkepanel__indikator"
              />
            </a>
            <a
              className="lenkepanel lenkepanel--border"
              href="http://localhost:9222/saksoversikt/"
            >
              <span
                className="typo-undertittel lenkepanel__heading"
              >
                <span>
                  Dine saker
                </span>
              </span>
              <span
                className="lenkepanel__indikator"
              />
            </a>
            <a
              className="lenkepanel last lenkepanel--border"
              href="http://localhost:9222/mininnboks/"
            >
              <span
                className="typo-undertittel lenkepanel__heading"
              >
                <span>
                  Innboks
                </span>
              </span>
              <span
                className="lenkepanel__indikator"
              />
            </a>
          </div>
          <h2
            className="typo-undertittel relatert-informasjon__subheader"
          >
            <span>
              Relatert informasjon
            </span>
          </h2>
          <div
            className="relatert-informasjon"
          >
            <nav
              className="relatert-informasjon__links"
            >
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="https://arbeidsplassen.nav.no/stillinger"
                >
                  Ledige stillinger
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/pselv/publisering/uforetrygd.jsf?context=ut"
                >
                  Uføretrygd
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="https://foreldrepenger.nav.no"
                >
                  Dine foreldrepenger
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/aktivitetsplan/"
                >
                  Aktivitetsplan
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/brukerprofil/"
                >
                  Din profil
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/veiledearbeidssoker/mistet-jobben/registrering"
                >
                  Registrer deg som arbeidssøker
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="https://stillingsok.nav.no/pam-stillingsok/lagrede-sok"
                >
                  Dine stillingssøk
                </a>
              </div>
              <div
                className="relatert-informasjon__link-container"
              >
                <a
                  className="lenke relatert-informasjon__link"
                  data-ga="Dittnav/Lenkeliste"
                  href="http://localhost:9222/person/personopplysninger"
                >
                  Personopplysninger
                </a>
              </div>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
